openscap-policies/src/main/scala/com/normation/plugins/openscappolicies/extension/OpenScapNodeDetailsExtension.scala

```diff
@@ -0,0 +1,87 @@
+package com.normation.plugins.openscappolicies.extension
+
+import com.normation.plugins.openscappolicies.OpenScapReport
+import com.normation.plugins.openscappolicies.services.{OpenScapReportReader, ReportSanitizer}
+import com.normation.plugins.{PluginExtensionPoint, PluginStatus}
+import com.normation.rudder.web.components.ShowNodeDetailsFromNode
+import net.liftweb.common.{EmptyBox, Full, Loggable}
+
+import scala.reflect.ClassTag
+import scala.xml.NodeSeq
+import net.liftweb.util.CssSel
+import net.liftweb.util.Helpers._
+import com.normation.box._
+import com.normation.inventory.domain.NodeId
+
+class OpenScapNodeDetailsExtension(
+    val status: PluginStatus
+  , openScapReader: OpenScapReportReader
+  , reportSanitizer: ReportSanitizer)(implicit val ttag: ClassTag[ShowNodeDetailsFromNode]) extends PluginExtensionPoint[ShowNodeDetailsFromNode] with Loggable {
+
+  def pluginCompose(snippet: ShowNodeDetailsFromNode) : Map[String, NodeSeq => NodeSeq] = Map(
+      "popupDetails" -> addOpenScapReportTab(snippet) _
+    , "mainDetails"  -> addOpenScapReportTab(snippet) _
+  )
+
+  /**
+   * Add a tab:
+   * - add an li in ul with id=openScapExtensionTab
+   */
+  def addOpenScapReportTab(snippet: ShowNodeDetailsFromNode)(xml: NodeSeq): NodeSeq = {
+    val content = openScapReader.checkOpenScapReportExistence(snippet.nodeId) match {
```

Comment:
here, perhaps we should check if the node has an openscap directive, and only in that case show the tab (and check report existence) ?